app/javascript/components/workflows/professor/exams/new/editor/components/questions/CodeTag.tsx

```diff
@@ -142,33 +143,77 @@ const FileModal: React.FC<FileModalProps> = (props) => {
   );
 };
 
-interface CodeTagProps {
-  qnum: number;
-  pnum: number;
-  bnum: number;
-  info: CodeTagInfo;
-  value: CodeTagState;
-  onChange: (newInfo: CodeTagInfo, newVal: CodeTagState) => void;
-  disabled: boolean;
-}
-
-const CodeTag: React.FC<CodeTagProps> = (props) => {
+const EditChoices: React.FC<WrappedFieldsProps> = (props) => {
+  const {
+    choices,
+    answer,
+  } = props;
+  const { input } = choices;
   const {
-    qnum,
-    pnum,
-    bnum,
-    info,
     value,
     onChange,
-    disabled,
+  }: {
+    value: CodeTagInfo['choices'];
+    onChange: (newVal: CodeTagInfo['choices']) => void;
+  } = input;
+  return (
+    <>
+      <Form.Label column sm={2}>Files source</Form.Label>
```

Comment:
Labels should have `for` fields (or `controlId` in the Form.Group, which indirectly does the same thing).  (To be clear, we've screwed this up elsewhere also, but we shouldn't introduce additional incomplete labels.)

Reply:
These are inside a form group with `controlId` set.